model/src/http/attachment.rs

```diff
@@ -2,48 +2,62 @@
 
 use serde::{Deserialize, Serialize};
 
-/// Attachment for when creating and updating messages.
+/// Attachments used in messages.
 ///
-/// `id` can be any placeholder value. If attaching multiple files to the
-/// message, each must be unique.
+/// # Examples
+///
+/// Create an attachment of a short JSON blob describing a cat with a
+/// description for screen readers:
+///
+/// ```
+/// use twilight_model::http::Attachment;
+///
+/// let filename = "twilight_sparkle.json".to_owned();
+/// let file_content = br#"{
+///     "best_friend": "Spike",
+///     "cutie_mark": "sparkles",
+///     "name": "Twilight Sparkle"
+/// }"#.to_vec();
+///
+/// let mut attachment = Attachment::from_bytes(filename, file_content, id);
+/// attachment.description("Raw data about Twilight Sparkle".to_owned());
+/// ```
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
 pub struct Attachment {
+    /// Description of the attachment, useful for screen readers and users
+    /// requiring alt text.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub description: Option<String>,
+    /// Content of the file.
     #[serde(skip)]
     pub file: Vec<u8>,
+    /// Name of the file.
+    ///
+    /// Examples may be "twilight_sparkle.png", "cat.jpg", or "logs.txt".
     pub filename: String,
+    /// Unique ID of the attachment in the message.
+    ///
+    /// All attachment IDs in a message must be unique, but may be any value of
```

Comment:
> must be unique

Should specify the uniqueness domain: global, guild, channel, this message or all messages of this user.